platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_tasks.adoc

```diff
@@ -130,3 +130,59 @@ include::sample[dir="snippets/bestPractices/useCacheabilityAnnotations/groovy",f
 === Tags
 
 `<<tags_reference.adoc#tag:tasks,#tasks>>`, `<<tags_reference.adoc#tag:caching,#caching>>`
+
+[[avoid_provider_get_outside_task_action]]
+== Do not call `get()`` on a Provider outside of a Task action
+:keywords: tasks,inputs-and-outputs
+
+When configuring tasks and extensions do not call link:{javadocPath}/org/gradle/api/provider/Provider.html#get()[`get()`] on a provider, use link:{javadocPath}/org/gradle/api/provider/Provider.html#map(org.gradle.api.Transformer)[`map()`], or link:{javadocPath}/https://docs.gradle.org/nightly/javadoc/org/gradle/api/provider/Provider.html#flatMap(org.gradle.api.Transformer)[`flatMap()`] instead
+
+=== Explanation
+
+A provider should be evaluated as late as possible. 
+Calling `get()` forces immediate evaluation, which can trigger unintended side effects, such as:
+
+- The value of the provider becomes an input to configuration, causing potential configuration cache misses.
+- The value may be evaluated too early, meaning you might not be using the final or correct value of the property. This may lead to painful and hard to debug ordering issues.
+- It breaks Gradle’s ability to build dependencies and to track task inputs and outputs, making automatic task dependency wiring impossible.
+
```

Comment:
🤔 I think this explanation needs a summary sentence at the end, after listing all the reasons not to trigger `get`.  Something like: "It is preferable to avoid explicitly evaluating a Provider at all, and deferring to `map/flatMap` to connect Providers to Providers implicitly."